File: 2552-count-increasing-quadruplets/2552-count-increasing-quadruplets.py
class Solution:
    def countQuadruplets(self, nums: List[int]) -> int:
        n = len(nums)
        dp = [0] * n
        ans = 0
        for i in range(n):
            prev = 0
            for j in range(i):
                if nums[j] < nums[i]:
                    prev += 1
                    ans += dp[j]
                else :
                    dp[j] += prev
        return ans

File: 2552-count-increasing-quadruplets/2552-count-increasing-quadruplets.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countQuadruplets = function(nums) {
    const n = nums.length
    const dp = Array(n).fill(0)
    let ans = 0
    for (let i = 0; i < n; i++)
        {
            let prev = 0
            for (let j = 0; j < i; j++){
                if (nums[j] > nums[i])
                    dp[j] += prev
                else {
                    ans += dp[j]
                    prev += 1
                }
                
            }
        }
    return ans
};